### PR TITLE
fix kube-dns CrashLoopBackOff issue with RBAC enabled minikube

### LINKED ISSUE #925: minikube should use RBAC
the handy `./tests/scripts/minikube.sh` script does not specify that minikube should use RBAC when it starts up minikube: https://github.com/rook/rook/blob/master/tests/scripts/minikube.sh#L58

It's useful to always be running with RBAC on so that we can catch any permissions regressions, new required permissions for new code, etc.

I thought this would be a simple fix to just pass `--extra-config=apiserver.Authorization.Mode=RBAC` as per https://github.com/kubernetes/minikube/blob/master/docs/configuring_kubernetes.md#examples.  This seems to work OK for Rook, but `kube-dns` is in a crash loop:
```
> kubectl -n kube-system get pod -l k8s-app=kube-dns
NAME                       READY     STATUS             RESTARTS   AGE
kube-dns-910330662-ccc9m   2/3       CrashLoopBackOff   53         2h
```

kubedns logs show that it's missing RBAC permissions:
```
> kubectl -n kube-system logs -lk8s-app=kube-dns -c kubedns
E0819 00:25:59.762761       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Endpoints: User "system:serviceaccount:kube-system:default" cannot list endpoints at the cluster scope. (get endpoints)
E0819 00:25:59.762857       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Service: User "system:serviceaccount:kube-system:default" cannot list services at the cluster scope. (get services)
E0819 00:25:59.763025       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.ConfigMap: User "system:serviceaccount:kube-system:default" cannot list configmaps in the namespace "kube-system". (get configmaps)
```

### PR DESCRIPTION
**Pull request by dangula**
[skip ci] fix #925

### CODE REVIEW

**Comment by galexrt** [comment]
@dangula Have you tried updating minikube to `v0.22.3`? After the `v0.22.x` update I didn't had this error anymore.

**Comment by dangula** [comment]
@galexrt I am using minikube v0.22.3.(with image minikube-v0.23.5.iso) And my kube-dns pod keeps restaring all the time. The kube-dns restart issue hasn't caused any issues but i thought i would be nice to fix it.

**Comment by galexrt** [comment]
@dangula I just noticed it happens again for me. Kube-dns working fine for me, was caused because I was using a minikube in which I already fixed the RBAC by hand earlier.

Yes, it is definitely good to fix this.

File: tests/scripts/minikube.sh [hunk 1]
```diff
@@ -16,6 +16,91 @@ EOF
     minikube start --memory=3000 --kubernetes-version ${KUBE_VERSION} --extra-config=apiserver.Authorization.Mode=RBAC
 }
 
+# workaround for kube-dns CrashLoopBackOff issue with RBAC enabled
+#issue https://github.com/kubernetes/minikube/issues/1734 and https://github.com/kubernetes/minikube/issues/1722
+enable_roles_for_RBAC() {
+cat <<EOF | kubectl create -f -
```
Comment (galexrt) [inline tests/scripts/minikube.sh]: Please add an indent here.

File: tests/scripts/minikube.sh [hunk 2]
```diff
@@ -16,6 +16,91 @@ EOF
...
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: cluster-writer
+rules:
+  - apiGroups: ["*"]
+    resources: ["*"]
+    verbs: ["*"]
+  - nonResourceURLs: ["*"]
+    verbs: ["*"]
+
```
Comment (galexrt) [inline tests/scripts/minikube.sh]: I would remove those empty lines in between the objects.

File: tests/scripts/minikube.sh (hunk 1 above)
Comment (dangula) [reply tests/scripts/minikube.sh]: done

File: tests/scripts/minikube.sh (hunk 2 above)
Comment (dangula) [reply tests/scripts/minikube.sh]: done

File: tests/scripts/minikube.sh [hunk 3]
```diff
@@ -16,6 +16,85 @@ EOF
     minikube start --memory=3000 --kubernetes-version ${KUBE_VERSION} --extra-config=apiserver.Authorization.Mode=RBAC
 }
 
+# workaround for kube-dns CrashLoopBackOff issue with RBAC enabled
+#issue https://github.com/kubernetes/minikube/issues/1734 and https://github.com/kubernetes/minikube/issues/1722
+enable_roles_for_RBAC() {
+ cat <<EOF | kubectl create -f -
```
Comment (galexrt) [inline tests/scripts/minikube.sh]: The indent from the other lines seems to be 4 spaces. Please change it here to look like the other lines.

Comment (dangula) [reply tests/scripts/minikube.sh]: oops fixed now.

File: tests/scripts/minikube.sh [hunk 4]
```diff
@@ -16,6 +16,85 @@ EOF
...
+    name: default
+    namespace: kube-system
+  - kind: User
+    name: kube-proxy
+roleRef:
+  kind: ClusterRole
+  name: cluster-writer
+  apiGroup: rbac.authorization.k8s.io
+---
+# Setup sd-build as a reader. This has to be a
```
Comment (jbw976) [inline tests/scripts/minikube.sh]: why do we need these `sd-*` accounts, roles and bindings?  in the original issue https://github.com/kubernetes/minikube/issues/1734, it looks like the sample fix came from the Screwdriver project: http://blog.screwdriver.cd/post/161863341372/set-up-screwdriver-in-kubernetes, which is where I think this `sd-*` stuff is coming from.

We aren't using screwdriver, so we shouldn't add those RBAC configs to our scripts, right?  I would *think* that only cluster-writer and cluster-reader are necessary.  Can you please try this scenario after removing all the `sd-*` stuff that I suspect isn't necessary?

Comment (dangula) [reply tests/scripts/minikube.sh]: thanks for catching that,i just coped the roles from the last comment in the issue. Cleaned them up now.

Comment (jbw976) [reply tests/scripts/minikube.sh]: @dangula can you confirm that kube-dns worked OK with RBAC enabled once you removed those `sd-*` permissions?

Comment (dangula) [reply tests/scripts/minikube.sh]: @jbw976 it worked, i ran full regession and all tests passed.

Comment (jbw976) [reply tests/scripts/minikube.sh]: ah, it looks like the cluster-reader role isn't used anymore now.  it was only used by the sd-build account.  let's get rid of the cluster-reader role too then I think this change will be complete.  thanks @dangula

**Pull request merged**